### PR TITLE
Add check for button text before rendering Button block

### LINKED ISSUE #17221: Button Block: Only display buttons if the title attribute exists
**Describe the bug**
Currently, a Button block without a `title` attribute still renders an empty button via the save function, leaving an empty `<a>` on the page. 

**To reproduce**
1. Add an empty Button block
2. Publish the page
3. View the published page

**Expected behavior**
A button is not rendered unless the `title` attribute is entered, preventing empty buttons from displaying if it's not intentionally configured within the editor.

**Screenshots**
ScreenFlow

**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser: Safari
 - Version: 12.1.1

**Additional context**
Using Gutenberg 6.3, although it happens regardless of whether or not the plugin is active.

### PR DESCRIPTION
**Pull request by richtabor**
## Description
This PR adds a check for the `text` attribute before rendering the Button block on the front-end. This prevents an empty `<div class="wp-block-button"><a class="wp-block-button__link"></a></div>` from displaying on the front-end. 

Closes #17221

## How has this been tested?
Tested with WP 5.5 RC
Tested to see if we need a deprecation (_not needed_)

### Before:
Editor:
Screen Shot 2020-07-20 at 12 18 40 PM
Front-end:
Screen Shot 2020-07-20 at 12 25 29 PM

### After:
Screen Shot 2020-07-20 at 12 18 40 PM
Screen Shot 2020-07-20 at 12 18 47 PM

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/button/save.js [hunk 1]
```diff
@@ -33,17 +33,21 @@ export default function save( { attributes } ) {
...
-				href={ url }
-				title={ title }
-				style={ buttonStyle }
-				value={ text }
-				target={ linkTarget }
-				rel={ rel }
-			/>
-		</div>
+		<>
+			{ text && (
```
Comment (ZebulanStanphill) [inline packages/block-library/src/button/save.js:37]: This could be simplified to just `return text && (...`. No need for a Fragment and `{`.

Comment (ZebulanStanphill) [reply packages/block-library/src/button/save.js:37]: Actually, I just had a better idea. Use an early return instead, for example:

```js
if ( ! text ) {
	return null;
}
```

Comment (ZebulanStanphill) [reply packages/block-library/src/button/save.js:37]: I just realized that the Fragment being the root element may actually be the cause of the test failures. The `wp-block-button` and `aligncenter` classes are not being applied to the `div` anymore.

**Comment by youknowriad** [comment]
This is probably going to require a deprecated version.

**Comment by ZebulanStanphill** [comment]
Why? As far as I can tell, this would have no effect on existing blocks until opening them in the editor, at which point they would update to the new markup upon save. The changes here (after applying my suggestions), should have no effect on what markup is considered a valid Button block.

**Comment by richtabor** [comment]
> This is probably going to require a deprecated version.

I tested for this as well but didn't see it needed one.

**Comment by youknowriad** [comment]
When you open an existing post on the editor, it performs validation to ensure markup was not updated externally. If it finds a different in the ouput compared to the saved markup, an invalidation issue is triggered. Almost any change to "save" functions of blocks requires a deprecated version. The thing that don't require are additions (with default values that don't impact markup).

I can't test now, but I'll circle back later to confirm.

**Comment by richtabor** [comment]
> When you open an existing post on the editor, it performs validation to ensure markup was not updated externally. If it finds a different in the ouput compared to the saved markup, an invalidation issue is triggered. Almost any change to "save" functions of blocks requires a deprecated version. The thing that don't require are additions (with default values that don't impact markup).
> 
> I can't test now, but I'll circle back later to confirm.

Yup, understood! I tested just for this - though maybe I missed something?

**Review by ZebulanStanphill — CHANGES_REQUESTED** [review]
[Switching to an early return (and dropping the Fragment use)](https://github.com/WordPress/gutenberg/pull/24064/files#r457620411) should fix the unit test failures (which are valid).

**Comment by youknowriad** [comment]
Here's a markup that you can build with master that will produce invalidation

```
<!-- wp:buttons -->
<div class="wp-block-buttons"><!-- wp:button -->
<div><a class="wp-block-button__link">test</a></div>
<!-- /wp:button -->

<!-- wp:button {"style":{"color":{"text":"#2c7ea1","background":"#6da2b9"}}} /--></div>
<!-- /wp:buttons -->
```

You were not experiencing the issue because if you don't apply custom colors, an existing deprecation (with isElligible) was picked up and it was upgrading the markup to the last version.

**Comment by getsource** [comment]
I added a deprecation and continued this in #29717.

**Comment by ZebulanStanphill** [comment]
Closing since #29717 was merged. Thanks for the initial work!

**Pull request closed without merging**